### ISSUE TITLE
Admin vs order locale potential mismatch

### ISSUE THREAD

**Issue by bitbager** [Sylius/Sylius#8897]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Sylius version   | 1.0.0

Hi! 
I have not tested this issue yet, but I was suggested on the Slack to open an issue to discuss this potential bug/feature request. 
I was wondering what would happen once an admin user sets the language preference to a different one that the order was placed with and then sends an email related to the order, for instance, shipping confirmation. I guess the email will be sent with the admin user's locale instead of the customer's one. Am I right?

**Comment by pjedrzejewski** [comment]
We record the language of the user during order placement and use it, so it should be fine. :) I think you have a bug in the issue's title. :D

**Comment by bitbager** [comment]
Sorry, misclick 😄 

I will try it tomorrow and let you know if it works as expected 🙂

**Comment by bitbager** [comment]
We checked it with the below case:

1. The order was placed with the fr locale.
2. Admin locale was set to en.
3. Shipment confirmation email was sent from the admin with EN translations.

Shall we consider it as a bug?

**Comment by GSadee** [comment]
If I think correctly, we aren't translating the emails, so it isn't a bug, but a feature request :)
[shipmentConfirmation.html.twig](https://github.com/Sylius/Sylius/blob/1.0/src/Sylius/Bundle/AdminBundle/Resources/views/Email/shipmentConfirmation.html.twig)
You can customize that in your project by overwriting the email templates and using a `localeCode` from an order.

**Comment by stefandoorn** [comment]
Sounds more like a bug than a feature request to me actually..

**Comment by bitbager** [comment]
@GSadee, there's also another problem that needs to be covered - once you want to display some dynamic content that's translatable, for ex. selected shipping method description, you can not do it with Twig's `|trans` filter which can take the locale as a parameter 🙁

**Comment by stefandoorn** [comment]
I'm doing something similar in a CLI command to generate a PDF and send email (unrelated to orders):

```
// Set locale to the language it got requested in (request in here is not an HTTP request btw)
$this->translator->setLocale($request->getLocale());
```

Afterwards `|trans` works in that language as far as I can see. It functions here in my command :)

**Comment by pamil** [comment]
Why `|trans` filter does not work? You can pass the locale as the third parameter.

**Comment by bitbager** [comment]
I will still not be able to force the translatable resources to present translated fields. For instance, `{{ shipment.method.description }}` in `shipmentConfirmation.html.twig`.

**Comment by pamil** [comment]
`{{ shipment.method.translation("locale").description }}` should work in that case.

**Comment by bitbager** [comment]
Yeap, looks like it should make the trick. But IMHO it's still a workaround. How about setting the email manager?

**Comment by Tetragramat** [comment]
Two years back I requested mailer pre_render event in #4329, so I could change locale. 
I think it would solve this issue, if we had pre and post render events, so we could change locale for rendering of the email. 
In my old sylius project I had to hardcode pre and post render events so I could change the locale. It worked perfectly. 
You cannot use parameters in `|trans`, because if there are multiple people working on the project, there is everytime someone who will forget to add locale parameter.
